### PR TITLE
006 Script for requirement [PerformAudioPassThru] requested "ttsChunks" is NOT supported

### PR DESCRIPTION
**Pull request by maria-mih**
006_ATF_P_PerformAudioPassThru_WARNINGS_Different_Speech_Capabilities_SAPI_PHONEMES.lua 
All common functions used available at #703
Testing coverage available at #643
@istoimenova Please review

### CODE REVIEW

**Comment by istoimenova** [comment]
@maria-mih, please reference script in the TestSet.

File: test_scripts/API/PerformAudioPassThru/006_ATF_P_PerformAudioPassThru_WARNINGS_Different_Speech_Capabilities_SAPI_PHONEMES.lua [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+-- In case mobile app sends PerformAudioPassThru_request to SDL with:
+-- valid <audioPassThruIcon> parameter
+-- and the requested <audioPassThruIcon> exists at app`s sandbox (see AppStorageFolder section)
+-- as well as another related to request valid mandatory params
+-- SDL must transfer UI.PerformAudioPassThru (<audioPassThruIcon>, other params)_request + Speak_request (depends on parameters provided by the app) to HMI
+--1. Used preconditions
+--1.1 All parameters are present and within bounds speech capabilities is sent with type = "SAPI_PHONEMES"
+--1.2. AudioPassThruIcon exists at apps sub-directory of AppStorageFolder (value from ini file)
+--2. Performed steps
+-- Send PerformAudioPassThru (audioPassThruIcon, other params, ttsChunksType = "SAPI_PHONEMES") from mobile to SDL and check:
```
Comment (istoimenova) [inline test_scripts/API/PerformAudioPassThru/006_ATF_P_PerformAudioPassThru_WARNINGS_Different_Speech_Capabilities_SAPI_PHONEMES.lua]: @maria-mih, please clarify why you test ttsChunksType = "SAPI_PHONEMES"?

File: test_scripts/API/PerformAudioPassThru/006_ATF_P_PerformAudioPassThru_WARNINGS_Different_Speech_Capabilities_SAPI_PHONEMES.lua [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+--2.1 SDL sends UI.PerformAudioPassThru (audioPassThruIcon, other params, ttsChunksType = "SAPI_PHONEMES") to HMI
```
Comment (istoimenova) [inline test_scripts/API/PerformAudioPassThru/006_ATF_P_PerformAudioPassThru_WARNINGS_Different_Speech_Capabilities_SAPI_PHONEMES.lua]: @maria-mih, ttsName(ttsChunksType) is parameter of TTS.Speak

File: test_scripts/API/PerformAudioPassThru/006_ATF_P_PerformAudioPassThru_WARNINGS_Different_Speech_Capabilities_SAPI_PHONEMES.lua [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+--2.2 SDL sends TTS.Speak to HMI
+--2.3 HMI sends UI.PerformAudioPassThru (SUCCESS) to SDL
```
Comment (istoimenova) [inline test_scripts/API/PerformAudioPassThru/006_ATF_P_PerformAudioPassThru_WARNINGS_Different_Speech_Capabilities_SAPI_PHONEMES.lua]: @maria-mih, points 2.3-2.4 are redundant in description.

File: test_scripts/API/PerformAudioPassThru/006_ATF_P_PerformAudioPassThru_WARNINGS_Different_Speech_Capabilities_SAPI_PHONEMES.lua [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+--2.4 HMI sends TTS.Speak (UNSUPPORTED_RESOURCE) to SDL
+-- Expected result:
```
Comment (istoimenova) [inline test_scripts/API/PerformAudioPassThru/006_ATF_P_PerformAudioPassThru_WARNINGS_Different_Speech_Capabilities_SAPI_PHONEMES.lua]: @istoimenova
istoimenova 6 minutes ago Member

istoimenova 3 minutes ago Member Pending

@maria-mih , expected result should be:
SDL sends UI.PerformAudioPassThru (audioPassThruIcon, other params) to HMI
SDL sends TTS.Speak to HMI

File: test_scripts/API/PerformAudioPassThru/006_ATF_P_PerformAudioPassThru_WARNINGS_Different_Speech_Capabilities_SAPI_PHONEMES.lua [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+-- SDL sends PerformAudioPassThru (WARNINGS, success:true) to mobile app
+
```
Comment (istoimenova) [inline test_scripts/API/PerformAudioPassThru/006_ATF_P_PerformAudioPassThru_WARNINGS_Different_Speech_Capabilities_SAPI_PHONEMES.lua]: @maria-mih, please remove empty row.

File: test_scripts/API/PerformAudioPassThru/006_ATF_P_PerformAudioPassThru_WARNINGS_Different_Speech_Capabilities_SAPI_PHONEMES.lua [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+---------------------------------------------------------------------------------------------
+
+--[[ General configuration parameters ]]
+config.deviceMAC = "12ca17b49af2289436f303e0166030a21e525d266e209267433801a8fd4071a0"
+
+--[[ Required Shared libraries ]]
+local commonFunctions = require ('user_modules/shared_testcases/commonFunctions')
```
Comment (istoimenova) [inline test_scripts/API/PerformAudioPassThru/006_ATF_P_PerformAudioPassThru_WARNINGS_Different_Speech_Capabilities_SAPI_PHONEMES.lua]: @maria-mih, please remove space before brackets. Apply this for all calls of functions / require.

File: test_scripts/API/PerformAudioPassThru/006_ATF_P_PerformAudioPassThru_WARNINGS_Different_Speech_Capabilities_SAPI_PHONEMES.lua [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+commonFunctions:newTestCasesGroup("Preconditions")
+
+commonSteps:PutFile("Precondition_PutFile_With_Icon","icon.png")
+
+function Test:Precondition_Check_audioPassThruIcon_Existence()
+  testCasesForPerformAudioPassThru:Check_audioPassThruIcon_Existence(self)
+end
+
+function Test:Precondition_ActivateApp()
+  testCasesForPerformAudioPassThru:ActivateAppDiffPolicyFlag
```
Comment (istoimenova) [inline test_scripts/API/PerformAudioPassThru/006_ATF_P_PerformAudioPassThru_WARNINGS_Different_Speech_Capabilities_SAPI_PHONEMES.lua]: @maria-mih, please move parameters on previous row.

File: test_scripts/API/PerformAudioPassThru/006_ATF_P_PerformAudioPassThru_WARNINGS_Different_Speech_Capabilities_SAPI_PHONEMES.lua [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+  (self, config.application1.registerAppInterfaceParams.appName, config.deviceMAC)
+end
+
+--[[ Test ]]
+commonFunctions:newTestCasesGroup("Test")
+function Test:TestStep_PerformAudioPassThru_PRE_SAPI_PHONEMES()
+  testCasesForPerformAudioPassThru:PerformAudioPassThru_Diff_Speech_Capabilities (self, "SAPI_PHONEMES")
```
Comment (istoimenova) [inline test_scripts/API/PerformAudioPassThru/006_ATF_P_PerformAudioPassThru_WARNINGS_Different_Speech_Capabilities_SAPI_PHONEMES.lua]: @maria-mih, as the idea of atomic script is to show what exactly is happening, will be better to declare the TestStep in script, not to call common function.
Calling of common functions is useful as preconditions/postconditions.

Comment (istoimenova) [inline test_scripts/API/PerformAudioPassThru/006_ATF_P_PerformAudioPassThru_WARNINGS_Different_Speech_Capabilities_SAPI_PHONEMES.lua]: @maria-mih , please be sure that RPC PerformAudioPassThru is allowed by policy.

File: test_scripts/API/PerformAudioPassThru/006_ATF_P_PerformAudioPassThru_WARNINGS_Different_Speech_Capabilities_SAPI_PHONEMES.lua (hunk 1 above)
Comment (maria-mih) [inline test_scripts/API/PerformAudioPassThru/006_ATF_P_PerformAudioPassThru_WARNINGS_Different_Speech_Capabilities_SAPI_PHONEMES.lua]: @istoimenova Agree that this is part of another requirement <<[PerformAudioPassThru] requested "ttsChunks" is NOT supported>>, description is updated accordingly.
Please check in commit 596dba59820a3fd836183ee41cb912bd6519ba31

File: test_scripts/API/PerformAudioPassThru/006_ATF_P_PerformAudioPassThru_WARNINGS_Different_Speech_Capabilities_SAPI_PHONEMES.lua (hunk 2 above)
Comment (maria-mih) [inline test_scripts/API/PerformAudioPassThru/006_ATF_P_PerformAudioPassThru_WARNINGS_Different_Speech_Capabilities_SAPI_PHONEMES.lua]: @istoimenova Updated at commit 596dba59820a3fd836183ee41cb912bd6519ba31

File: test_scripts/API/PerformAudioPassThru/006_ATF_P_PerformAudioPassThru_WARNINGS_Different_Speech_Capabilities_SAPI_PHONEMES.lua (hunk 3 above)
Comment (maria-mih) [inline test_scripts/API/PerformAudioPassThru/006_ATF_P_PerformAudioPassThru_WARNINGS_Different_Speech_Capabilities_SAPI_PHONEMES.lua]: @istoimenova Updated at commit 596dba59820a3fd836183ee41cb912bd6519ba31

File: test_scripts/API/PerformAudioPassThru/006_ATF_P_PerformAudioPassThru_WARNINGS_Different_Speech_Capabilities_SAPI_PHONEMES.lua (hunk 4 above)
Comment (maria-mih) [inline test_scripts/API/PerformAudioPassThru/006_ATF_P_PerformAudioPassThru_WARNINGS_Different_Speech_Capabilities_SAPI_PHONEMES.lua]: @istoimenova Updated at commit 596dba59820a3fd836183ee41cb912bd6519ba31

File: test_scripts/API/PerformAudioPassThru/006_ATF_P_PerformAudioPassThru_WARNINGS_Different_Speech_Capabilities_SAPI_PHONEMES.lua (hunk 5 above)
Comment (maria-mih) [inline test_scripts/API/PerformAudioPassThru/006_ATF_P_PerformAudioPassThru_WARNINGS_Different_Speech_Capabilities_SAPI_PHONEMES.lua]: @istoimenova Updated at commit 596dba59820a3fd836183ee41cb912bd6519ba31

File: test_scripts/API/PerformAudioPassThru/006_ATF_P_PerformAudioPassThru_WARNINGS_Different_Speech_Capabilities_SAPI_PHONEMES.lua (hunk 6 above)
Comment (maria-mih) [inline test_scripts/API/PerformAudioPassThru/006_ATF_P_PerformAudioPassThru_WARNINGS_Different_Speech_Capabilities_SAPI_PHONEMES.lua]: @istoimenova Updated at commit 596dba59820a3fd836183ee41cb912bd6519ba31

File: test_scripts/API/PerformAudioPassThru/006_ATF_P_PerformAudioPassThru_WARNINGS_Different_Speech_Capabilities_SAPI_PHONEMES.lua (hunk 7 above)
Comment (maria-mih) [inline test_scripts/API/PerformAudioPassThru/006_ATF_P_PerformAudioPassThru_WARNINGS_Different_Speech_Capabilities_SAPI_PHONEMES.lua]: @istoimenova Updated at commit 596dba59820a3fd836183ee41cb912bd6519ba31

File: test_scripts/API/PerformAudioPassThru/006_ATF_P_PerformAudioPassThru_WARNINGS_Different_Speech_Capabilities_SAPI_PHONEMES.lua (hunk 8 above)
Comment (maria-mih) [inline test_scripts/API/PerformAudioPassThru/006_ATF_P_PerformAudioPassThru_WARNINGS_Different_Speech_Capabilities_SAPI_PHONEMES.lua]: @istoimenova Updated at commit 596dba59820a3fd836183ee41cb912bd6519ba31

Comment (maria-mih) [inline test_scripts/API/PerformAudioPassThru/006_ATF_P_PerformAudioPassThru_WARNINGS_Different_Speech_Capabilities_SAPI_PHONEMES.lua]: @istoimenova Updated at commit 596dba59820a3fd836183ee41cb912bd6519ba31

**Comment by maria-mih** [comment]
@istoimenova Script has been updated after first review iteration. Please check on commit 596dba59820a3fd836183ee41cb912bd6519ba31

**Pull request merged**